tldw_Server_API/app/core/MCP_unified/modules/implementations/flashcards_module.py

```
@@ -447,6 +447,20 @@ def _open_db(self, context: Any) -> CharactersRAGDB:
             raise ValueError("ChaChaNotes DB path not available in context")
         return CharactersRAGDB(db_path=chacha_path, client_id=f"mcp_flashcards_{self.config.name}")
 
+    def _workspace_id_from_context(self, context: Any | None) -> str | None:
```

Ruff (0.15.9)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `context`

(ANN401)